src/components/EditCoursePage/index.jsx

```diff
@@ -111,7 +118,52 @@ class EditCoursePage extends React.Component {
     return updatedCourseEditData;
   }
 
-  handleCourseEdit(courseData) {
+  showModal(submitCourseData) {
+    const {
+      targetRun,
+    } = this.state;
+
+    if (targetRun && !(targetRun.status === PUBLISHED)) {
+      // Submitting Run for review, show modal, and temporarily store form data until
+      // we have a response for how to continue.
+      this.setState({
+        submitCourseData,
+        submitConfirmVisible: true, // show modal
+      });
+    } else {
+      // Edit submission, no modal required.
+      this.handleCourseEdit(submitCourseData);
+    }
+  }
+
+  cancelSubmit() {
+    const {
+      clearSubmitStatus,
+    } = this.props;
+
+    this.setState({
+      submitCourseData: {},
+      submitConfirmVisible: false,
+    });
+
+    store.dispatch(stopSubmit(this.getFormId()));
```

Comment:
I think you removed this since we didn't need it but just commenting for completeness